x-pack/plugins/security_solution/public/dashboards/containers/use_fetch_security_tags.ts

```diff
@@ -9,12 +9,19 @@ import { useEffect, useMemo, useRef } from 'react';
 import { useKibana } from '../../common/lib/kibana';
 import { createTag, getTagsByName } from '../../common/containers/tags/api';
 import { REQUEST_NAMES, useFetch } from '../../common/hooks/use_fetch';
-import { SECURITY_TAG_DESCRIPTION, SECURITY_TAG_NAME } from '../../../common/constants';
-import { getRandomColor } from '../../../common/utils/get_ramdom_color';
+import {
+  SECURITY_TAG_DESCRIPTION,
+  SECURITY_TAG_NAME,
+  getSecurityTagId,
+} from '../../../common/constants';
+import { getRandomColor } from '../../../common/utils/get_random_color';
+import { useSpaceId } from '../../common/hooks/use_space_id';
 
 export const useFetchSecurityTags = () => {
   const { http, savedObjectsTagging } = useKibana().services;
   const tagCreated = useRef(false);
+  const spaceId = useSpaceId() ?? 'default';
+  const defaultSecuritySolutionTagId = getSecurityTagId(spaceId);
```

Comment:
I think `useSpaceId() ?? 'default'` is a potentially error-prone expression, because when you're on a non-default space, but space id inside the `useSpaceId` hook hasn't been loaded yet, you will get a `-default` tag instead of the correct one.